README.md

```diff
@@ -130,29 +128,80 @@ Examples:
 
 ### Scopes
 
-Scopes should be specified in disjunctive normal form.  In other words, an
-array of arrays, where all of the scopes in at least one of the inner arrays
-must be satisfied.  Parameters are substituted into scopes with `<paramName>`
+Scopes should be specified in
+[scope expression form](https://github.com/taskcluster/taskcluster-lib-scopes#new-style).
+Parameters are substituted into scopes with `<paramName>`
 syntax.  For example, the following defintion allows the method when *either*
 the caller's scopes satisfy `queue:create-task..` for the given `provisionerId`
 and `workerType`, *or* the caller's scopes satisfy all of
 `queue:define-task:..`, `queue:task-group-id:..`, and `queue:schedule-task:..`.
 
 ```js
-  scopes: [
-    [
+  scopes: {AnyOf: [
+    {AllOf: [
       'queue:create-task:<provisionerId>/<workerType>',
-    ], [
+    ]},
+    {AllOf: [
       'queue:define-task:<provisionerId>/<workerType>',
       'queue:task-group-id:<schedulerId>/<taskGroupId>',
       'queue:schedule-task:<schedulerId>/<taskGroupId>/<taskId>',
-    ],
-  ],
+    ]},
+  ]},
 ```
 
 If scope validation fails, the user is presented with an extensive error
 message indicating the available and required scopes.
 
+Arrays of required scopes can also be templated in the scopes section. When
+this form is used, the parameter must be an array.
+
+```js
+  scopes: {AllOf: [
+    {for: 'route', in: 'routes', each: 'queue:route:<route>'}
+  ]}
+```
+
+Given a parameter `routes` that is an array of strings as in `['foo', 'bar']`,
+this will evaluate to:
+
+```js
+  scopes: {AllOf: ['queue:route:foo', 'queue:route:bar']}
+```
+
+You may also use if/then constructs in scope expressions. In this case, the `if`
+field should be a parameter and the `then` must be a scope expression that will
+be subsituted in for the if/then block if the `if` parameter exists and is truthy.
+Given the follwoing example:
+
+```js
+  scopes: {AnyOf: ['foo:bar', {if: 'public', then: {AnyOf: []}}]}
```

Comment:
I would change this example to be:
```js
scopes: {AllOf: [{if: 'not_public', then: 'foo:bar'}]}
```

Reply:
Surely there's room for a wee `!` operator in there?